lib/content-services/src/lib/search/components/search-filter/search-filter.component.ts

```diff
@@ -397,12 +397,12 @@ export class SearchFilterComponent implements OnInit, OnDestroy {
         });
     }
 
-    private getBucketFilterFunction(bucketList) {
+    protected getBucketFilterFunction(bucketList) {
         return (bucket: FacetFieldBucket): boolean => {
             if (bucket && bucketList.filterText) {
                 const pattern = (bucketList.filterText || '').toLowerCase();
                 const label = (this.translationService.instant(bucket.display) || this.translationService.instant(bucket.label)).toLowerCase();
-                return this.queryBuilder.config.filterWithContains ? label.indexOf(pattern) !== -1 : label.startsWith(pattern);
+                return this.queryBuilder.config.filterWithContains ? label.indexOf(pattern) !== -1 : label.includes(pattern);
```

Comment:
what's the difference then compared to the other condition and `filterWithContains` ?

Reply:
What other condition ? Sorry I don't understand? :/
are you talking about `label.indexOf(pattern)` ? I don't know but which is for sure, only the "starts with" works. Currently, if you search for "A" in a list [Type A, Type B, Type C], the search does not work. You have to search for "Type A" to get a result.
We already have 4 customers who are against this. 
If the method had been "protected" I would have been able to correct the problem only from our side without disturbing you :/

Reply:
The "includes" is essentially doing the "contains" comparison, the "indexOf" is also checking for "contains", so the code logic can be read as:
```
if filterWithContains === true then check contains
else check contains
```

Reply:
![image](https://user-images.githubusercontent.com/8958727/75237262-584c8000-57bf-11ea-863f-7356d727fd07.png)

Reply:
I think you have fixed the wrong condition branch, might be 
```
return this.queryBuilder.config.filterWithContains 
  ? label.includes(pattern)
  : label.startsWith(pattern);
```

Reply:
try to enable the `filterWithContains` in the config. I think you are trying to fix a non-existing problem

Reply:
OK, I understand ! I didn't add "filterWithContains" in "app.config.json" !  
Sorry :(  It's actually working.

Reply:
no worries, I'm glad it works for you